conans/client/build/cmake.py

```diff
@@ -289,12 +289,19 @@ def build(self):
                 cmake.install()
                 cmake.patch_config_paths()
         """
+
+        def _format_cmake_config_path(pathstr):
+            if self._conanfile.settings.os == "Windows":
```

Comment:
I'm not sure I understand - I see the test failure (don't quite understand it yet) but this is a package build time issue only. We don't have any way at the moment of live patching a cmake config file from a _consumer_ (might be a good idea, IDK ??). However, I'm also not remotely sure what happens when you generate a cmake export config in a cross-building from windows to linux environment... I think perhaps it might be better to check first if the settings has os, then theck the os...

Reply:
on the other hand, we also need to consider if the way this test is written is correct... I suppose actually, it is possible that a header only package (i.e. one which might not have an "os" setting) would export a cmake config file, and therefore we should be testing the os of the system it's built on...

Reply:
That is exactly what I mean. The setting `os` in a header only could not exist, but probably you want to still patch the package, right? And probably if you are creating the package in Windows (platform.system() == "Windows") is where you need to replace the chars. The consumer is not related, this method is run when the package is built, and it matters in which system is being built rather than the system that is targeting.